src/libs/actions/Transaction.js

```diff
@@ -31,4 +52,49 @@ function addStop(transactionID, newLastIndex) {
     });
 }
 
-export {addStop, createInitialWaypoints};
+/**
+ * Saves the selected waypoint to the transaction
+ * @param {String} transactionID
+ * @param {String} index
+ * @param {Object} waypoint
+ */
+function saveWaypoint(transactionID, index, waypoint) {
+    Onyx.merge(`${ONYXKEYS.COLLECTION.TRANSACTION}${transactionID}`, {
+        comment: {
+            waypoints: {
+                [`waypoint${index}`]: waypoint,
+            },
+        },
+    });
+}
+
+function removeWaypoint(transactionID, currentIndex) {
+    // Index comes from the route params and is a string
+    const index = Number(currentIndex);
+    const transaction = lodashGet(allTransactions, transactionID, {});
+    const existingWaypoints = lodashGet(transaction, 'comment.waypoints', {});
+    const totalWaypoints = _.size(existingWaypoints);
+
+    // Prevents removing the starting or ending waypoint but clear the stored address
+    if (index === 0 || index === totalWaypoints - 1) {
```

Comment:
If I have three waypoints (start, stop, finish) I should be able to remove the finish waypoint which will make the old stop the new finish. Similarly for the start waypoint.

We could add to the beginning of this `if(totalWaypoints === 2 && ...`